mirage/unikernel.ml

```diff
@@ -445,81 +506,126 @@ module Main (N : Mirage_net.S) (ASSETS : Mirage_kv.RO) = struct
 
   type t = { net : Net.t; mac : Macaddr.t; mutable configuration : string }
 
-  let dhcp_configuration_of mac dhcp_range domain =
-    let v4_address = Ipaddr.V4.Prefix.address (K.ipv4 ()) in
+  let process_config configuration =
+    let ( let* ) = Result.bind in
+    let unique x item =
+      match x with
+      | None -> Ok ()
+      | Some _ -> Error (item ^ " must only appear once")
+    in
+    let rec gather acc : Dnsvizor.Config_parser.config -> _ = function
+      | [] -> Ok acc
+      | `Ignored :: r -> gather acc r
+      | `Dhcp_range
+          ( { mode = Some _; _ }
+          | { broadcast = Some _; _ }
+          | { lease_time = Some _; _ }
+          | { netmask = Some _; _ } )
+        :: _ ->
+          Error "Unhandled dhcp-range option"
```

Comment:
I'm not sure what is handled/not handled here? anything with mode, broadcast, lease_time, or netmask is not handled?